packages/ui/src/components/button.tsx

```diff
@@ -1,169 +1,83 @@
-        success:
-          'border-button-border-success-1 bg-button-background-success-1 text-button-foreground-success-1 hover:bg-button-background-success-2',
-        muted: 'border-cn-borders-3/20 bg-cn-background-1/10 text-cn-foreground-3',
-        primary: 'border-cn-borders-brand/20 bg-cn-background-primary text-cn-foreground-primary',
-        disabled:
-          'border-button-border-disabled-1 bg-button-background-disabled-1 text-button-foreground-disabled-1 disabled:bg-button-background-disabled-1 disabled:text-button-foreground-disabled-1'
-      },
-      padding: {
-        default: '',
-        sm: 'px-3'
-      },
-      gradient: {
-        default: 'ai-button'
-      }
+import { Icon } from './icon'
+
+const buttonVariants = cva('button', {
+  variants: {
+    variant: {
+      solid: 'button-solid',
+      surface: 'button-surface',
+      soft: 'button-soft',
+      ghost: 'button-ghost',
+      link: 'button-link'
+    },
+    size: {
+      sm: 'button-sm',
+      lg: 'button-lg'
+    },
+    rounded: {
+      true: 'button-rounded'
+    },
+
+    iconOnly: {
+      true: 'button-icon-only'
     },
-    defaultVariants: {
-      variant: 'default',
-      size: 'default',
-      theme: 'default',
-      padding: 'default',
-      borderRadius: 'default',
-      gradient: 'default'
+
+    theme: {
+      success: 'button-success',
+      muted: 'button-muted',
+      danger: 'button-danger',
+      primary: 'button-primary',
+      ai: 'button-ai'
     }
+  },
+  defaultVariants: {
+    variant: 'solid',
+    theme: 'primary'
   }
-)
+})
 
 interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement>, VariantProps<typeof buttonVariants> {
   asChild?: boolean
   loading?: boolean
-  spinner?: ReactNode
-  dropdown?: ReactNode
-  gradientType?: string
+  rounded?: boolean
+  iconOnly?: boolean
 }
 
+export type ButtonThemes = Exclude<NonNullable<VariantProps<typeof buttonVariants>['theme']>, null | undefined>
+export type ButtonVariants = Exclude<NonNullable<VariantProps<typeof buttonVariants>['variant']>, null | undefined>
+export type ButtonSizes = VariantProps<typeof buttonVariants>['size']
+
+// add icon only aria attr if iconOnly is true
+
 const Button = forwardRef<HTMLButtonElement, ButtonProps>(
   (
-    {
-      className,
-      variant,
-      size,
-      padding,
-      theme,
-      borderRadius,
-      gradientType,
-      asChild = false,
-      loading,
-      disabled,
-      spinner,
-      dropdown,
-      children,
-      ...props
-    },
+    { className, variant, size, theme, rounded, iconOnly, asChild = false, loading, disabled, children, ...props },
     ref
   ) => {
     const Comp = asChild ? Slot : 'button'
 
     const _children = loading ? (
       <>
-        {(loading && spinner) ||
-          CanaryOutletFactory.getOutlet(CanaryOutletName.BUTTON_SPINNER, {
```

Comment:
I removed the outlet for spinner and I added static spinner. 

cc: @abhinavrastogi-harness